src/Shared/LayoutRenderers/AspNetRequestUrlRenderer.cs

```diff
@@ -89,56 +169,74 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
 
         private void RenderUrl(HttpRequestBase httpRequest, StringBuilder builder)
         {
+            if (Properties == AspNetRequestUrlProperty.None)
+            {
+                return;
+            }
+
             var url = httpRequest.Url;
             if (url == null)
+            {
                 return;
+            }
 
-            if (IncludeScheme && !string.IsNullOrEmpty(url.Scheme))
+            // We cannot use Enum.HasFlag(Enum) in .NET 35
+            if ((Properties & AspNetRequestUrlProperty.Scheme) == AspNetRequestUrlProperty.Scheme && 
+                !string.IsNullOrEmpty(url.Scheme))
             {
                 builder.Append(url.Scheme);
                 builder.Append("://");
             }
-            if (IncludeHost)
+            if ((Properties & AspNetRequestUrlProperty.Host) == AspNetRequestUrlProperty.Host)
             {
                 builder.Append(url.Host);
             }
-            if (IncludePort && url.Port > 0)
+            if ((Properties & AspNetRequestUrlProperty.Port) == AspNetRequestUrlProperty.Port && url.Port > 0)
             {
                 builder.Append(':');
                 builder.Append(url.Port);
             }
 
-            if (IncludePath)
+            if ((Properties & AspNetRequestUrlProperty.Path) == AspNetRequestUrlProperty.Path)
             {
-                var pathAndQuery = IncludeQueryString ? url.PathAndQuery : url.AbsolutePath;
+                var pathAndQuery = (Properties & AspNetRequestUrlProperty.QueryString) == AspNetRequestUrlProperty.QueryString ? 
+                    url.PathAndQuery : url.AbsolutePath;
                 builder.Append(pathAndQuery);
             }
-            else if (IncludeQueryString)
+            else if ((Properties & AspNetRequestUrlProperty.QueryString) == AspNetRequestUrlProperty.QueryString)
             {
                 builder.Append(url.Query);
             }
         }
 #else
         private void RenderUrl(HttpRequest httpRequest, StringBuilder builder)
         {
-            if (IncludeScheme && !string.IsNullOrWhiteSpace(httpRequest.Scheme))
+            if (Properties == AspNetRequestUrlProperty.None)
+            {
+                return;
+            }
+
+            // We can use Enum.HasFlag(Enum) in .NET 4.0 and higher
+            if ( Properties.HasFlag(AspNetRequestUrlProperty.Scheme) && 
```

Comment:
Very happy that you used the old-style bit-wise-operation for `ASP_NET_CORE`. Can the same be done here also, instead of using `HasFlag`.

Reply:
Yes that is cool